statsd-influxdb-backend-v0.12

```diff
@@ -0,0 +1,641 @@
+    }
+  });
+
+  req.on('error', function (e, i) {
+    self.log(e);
+  });
+
+  var payload = JSON.stringify(points)
+  self.influxdbStats.payloadSize = Buffer.byteLength(payload);
+
+  self.logDebug(function () {
+    var size = (self.influxdbStats.payloadSize / 1024).toFixed(2);
+    return 'Payload size ' + size + ' KB';
+  });
+
+  req.write(payload);
+  req.end();
+}
+
+InfluxdbBackend.prototype.httpPOST_v09 = function (points) {
+  /* Do not send if there are no points. */
+  if (!points.length) { return; }
+
+  var self = this,
+      query = {u: self.user, p: self.pass},
+      protocolName = self.protocol == http ? 'HTTP' : 'HTTPS',
+      startTime;
+
+  self.logDebug(function () {
+    return 'Sending ' + points.length + ' different points via ' + protocolName;
+  });
+
+  self.influxdbStats.numStats = points.length;
+
+  var options = {
+    hostname: self.host,
+    port: self.port,
+    path: '/write?' + querystring.stringify(query),
+    method: 'POST',
+    agent: false // Is it okay to use "undefined" here? (keep-alive)
+  };
+
+  var req = self.protocol.request(options);
+
+  req.on('socket', function (res) {
+    startTime = process.hrtime();
+  });
+
+  req.on('response', function (res) {
+    var status = res.statusCode;
+
+    self.influxdbStats.httpResponseTime = millisecondsSince(startTime);
+
+    if (status >= 400) {
+      self.log(protocolName + ' Error: ' + status);
+    }
+  });
+
+  req.on('error', function (e, i) {
+    self.log(e);
+  });
+
+  var payload = JSON.stringify({
+    database: self.database,
+    points: points
+  });
+
+  self.influxdbStats.payloadSize = Buffer.byteLength(payload);
+
+  self.logDebug(function () {
+    var size = (self.influxdbStats.payloadSize / 1024).toFixed(2);
+    return 'Payload size ' + size + ' KB';
+  });
+
+  req.write(payload);
+  req.end();
+} //end of InfluxdbBackend.prototype.httpPOST_v09
+
+	
+InfluxdbBackend.prototype.assembleEvent_v11 = function (name, events) {
+  var self = this;
+
+//  var payload = {
+//    measurement: name,
+//    fields: { value: events[0]['value'] }
+//  }
+  var payload=name+" value="+events[0]['value']+" "+events[0]['time']*1000000;
+  return payload;
+}
+/***
+ * @ Vita Zhao add new version httpPOST of version 0.11 with http call
+ */
+InfluxdbBackend.prototype.httpPOST_v11 = function (points) {
+  /* Do not send if there are no points. */
+  if (!points.length) { return; }
+
+  var self = this,
+      query = {u: self.user, p: self.pass},
+      protocolName = self.protocol == http ? 'HTTP' : 'HTTPS',
+      startTime;
+
+  self.logDebug(function () {
+    return 'Sending ' + points.length + ' different points via ' + protocolName;
+  });
+
+  self.influxdbStats.numStats = points.length;
+
+  var options = {
+    hostname: self.host,
+    port: self.port,
+    //path: '/write?' + querystring.stringify(query),//TODO:/write?db=dbname
+    path: '/write?db=' + self.database+'&rp='+self.retentionPolicy,//TODO:/write?db=dbname
```

Comment:
This will throw a 401 when using HTTP authentication since we're not passing the username and password from `query`